botocore/utils.py

```diff
@@ -980,13 +980,13 @@ def _get_response(self, full_url, headers, timeout):
             if response.status_code != 200:
                 raise MetadataRetrievalError(
                     error_msg="Received non 200 response (%s) from ECS metadata: %s"
-                    % (response.status_code, response.content))
+                    % (response.status_code, response.text))
             try:
-                return json.loads(response.content)
+                return json.loads(response.content.decode('utf-8'))
```

Comment:
Given your rationale for using `response.text` in the PR description, seems like using `response.text` here also would make sense.